addons/base_import_module/models/ir_module.py

```diff
@@ -5,21 +6,21 @@
 from os.path import join as opj
 
 import openerp
-from openerp.osv import osv
+from openerp import _, models
 from openerp.tools import convert_file
-from openerp.tools.translate import _
 from openerp.exceptions import UserError
 
 _logger = logging.getLogger(__name__)
 
 MAX_FILE_SIZE = 100 * 1024 * 1024 # in megabytes
 
-class view(osv.osv):
+
+class View(models.Model):
     _inherit = "ir.module.module"
 
-    def import_module(self, cr, uid, module, path, force=False, context=None):
-        known_mods = self.browse(cr, uid, self.search(cr, uid, []))
-        known_mods_names = dict([(m.name, m) for m in known_mods])
+    def import_module(self, module, path, force=False):
+        known_mods = self.search([])
```

Comment:
May be you are partially right, known_mods are further used to get module which going to import.